src/ln/channel.rs

```diff
@@ -1653,21 +1653,35 @@ impl Channel {
 
 		let local_keys = self.build_local_transaction_keys(self.cur_local_commitment_transaction_number)?;
 
+		let mut update_fee = false;
 		let feerate_per_kw = if !self.channel_outbound && self.pending_update_fee.is_some() {
+			update_fee = true;
 			self.pending_update_fee.unwrap()
 		} else {
 			self.feerate_per_kw
 		};
 
+
 		let mut local_commitment_tx = self.build_commitment_transaction(self.cur_local_commitment_transaction_number, &local_keys, true, false, feerate_per_kw);
 		let local_commitment_txid = local_commitment_tx.0.txid();
 		let local_sighash = Message::from_slice(&bip143::SighashComponents::new(&local_commitment_tx.0).sighash_all(&local_commitment_tx.0.input[0], &funding_script, self.channel_value_satoshis)[..]).unwrap();
 		secp_call!(self.secp_ctx.verify(&local_sighash, &msg.signature, &self.their_funding_pubkey.unwrap()), "Invalid commitment tx signature from peer", self.channel_id());
 
+		//If channel fee was updated by funder confirm funder can afford the new fee rate when applied to the current local commitment transaction
+		if update_fee {
+			let num_htlcs = local_commitment_tx.0.output.len();
```

Comment:
num_htlcs should be output.len() - 2, no? The test should probably catch this. Note that build_commitment_transaction is somewhat confusingly written - the fees are calculated before the non-HTLC outputs are added to the txouts Vec.

Reply:
Ahh! This did confuse me! I thought that the way I had setup the network that we had only done the initial funding and a few updates so I expected there to be no HTLCs in the channel yet but my testing revealed there was 1. So I think it will actually be `max(outout.len() - 2, 0)` because if one of the non-HTLC outputs is below the dust limit it is not added to the txn.

Reply:
I actually think I might have to check the value_to_a and value_to_b in commitment_signed to see if either is below the dust limit to know how many to deduct from the output.len(). I don't see a way to differentiate the HTLC and non-HTLC outputs from the txn without looking at the scripts which seems more intensive.

Reply:
Actually I should probably just use the length of Vec<HTLCOutputInCommitment>.